app/assets/javascripts/discourse/app/initializers/welcome-topic-banner.js

```diff
@@ -0,0 +1,16 @@
+export default {
+  name: "welcome-topic-banner",
+  after: "message-bus",
+
+  initialize(container) {
+    const messageBus = container.lookup("service:message-bus");
+    if (!messageBus) {
+      return;
+    }
```

Comment:
I'm not sure we need this conditional here? This is an Ember service object so it can never be undefined or null.

Reply:
I've seen this in in multiple existing implementations for example: [`logout.js`](https://github.com/discourse/discourse/blob/067c4deb4c34ad5100e51e91bf0668d125de7908/app/assets/javascripts/discourse/app/initializers/logout.js#L13-L17), [`read-only.js`](https://github.com/discourse/discourse/blob/067c4deb4c34ad5100e51e91bf0668d125de7908/app/assets/javascripts/discourse/app/initializers/read-only.js#L7-L10), [`banner.js`](https://github.com/discourse/discourse/blob/067c4deb4c34ad5100e51e91bf0668d125de7908/app/assets/javascripts/discourse/app/initializers/banner.js#L14-L17).

Reply:
Those are lines which have been added 7 years and I'm pretty sure it isn't relevant today. If we look at the MessageBus service, I don't think the service will ever be `null` or `undefined`.

https://github.com/discourse/discourse/blob/0d4b1f25f0c5c0cb292a81103ad49905ab02ba45/app/assets/javascripts/discourse/app/services/message-bus.js#L3-L9